src/Glpi/Inventory/Inventory.php

```diff
@@ -287,7 +287,7 @@ public function doInventory($test_rules = false)
 
         $main_start = microtime(true); //bench
         try {
-            if (!$DB->inTransaction()) {
+            if (!defined('TU_USER')) {
```

Comment:
The commit operation a bit lower in this method was already conditioned on TU_USER.

I tried to remove it but it then breaks all tests so we must keep it for now and apply the same check for the beginTransaction and the rollback operations.